src/main.cpp

```diff
@@ -6106,7 +6106,8 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
             LOG(NET, "more getheaders (%d) to end to peer=%s (startheight:%d)\n", pindexLast->nHeight,
                 pfrom->GetLogName(), pfrom->nStartingHeight);
             pfrom->PushMessage(NetMsgType::GETHEADERS, chainActive.GetLocator(pindexLast), uint256());
-
+            CNodeState* state = State(pfrom->GetId());
+            if (state) state->nSyncStartTime = GetTime();  // reset the time because more headers needed
             // During the process of IBD we need to update block availability for every connected peer. To do that we
```

Comment:
nit -> leave a newline after this ...

Also need a dbgassert here

Reply:
plus the lock on `cs_main` like requested for line 6423?

Reply:
cs_main is already locked further up in this instance.

Reply:
fair enough